### DISCUSSION TITLE
Run pipeline with two API keys

### DISCUSSION THREAD

**Discussion by C-Roensholt** [dagster-io/dagster#18988] [Q&A] [answered]
We have a system where men and women have separate accounts. This means that we have two API keys for the same system and thus pipeline.
What is the best and most efficient way to run the the same pipeline (assets) with two different API keys (resources)?

**Comment by clairelin135** [comment, marked as answer]
Is it possible to define your resource such that it accepts the API key at run launch time via resource config?

If so, the following may work for you: https://docs.dagster.io/concepts/resources#configuring-resources-at-launch-time

**Reply by C-Roensholt** [reply]
This looks like a good way to address it.
I will explore it, thank you!
